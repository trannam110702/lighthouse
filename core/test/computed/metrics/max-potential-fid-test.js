/**
 * @license
 * Copyright 2018 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import {MaxPotentialFID} from '../../../computed/metrics/max-potential-fid.js';
import {getURLArtifactFromDevtoolsLog, readJson} from '../../test-utils.js';

const trace = readJson('../../fixtures/artifacts/progressive-app/trace.json', import.meta);
const devtoolsLog = readJson('../../fixtures/artifacts/progressive-app/devtoolslog.json', import.meta);

const URL = getURLArtifactFromDevtoolsLog(devtoolsLog);

describe('Metrics: Max Potential FID', () => {
  const gatherContext = {gatherMode: 'navigation'};

  it('should compute a simulated value', async () => {
    const settings = {throttlingMethod: 'simulate'};
    const context = {settings, computedCache: new Map()};
    const result = await MaxPotentialFID.request({trace, devtoolsLog, gatherContext, settings, URL},
      context);

    expect({
      timing: Math.round(result.timing),
      optimistic: Math.round(result.optimisticEstimate.timeInMs),
      pessimistic: Math.round(result.pessimisticEstimate.timeInMs),
    }).toMatchSnapshot();
  });

  it('should compute an observed value', async () => {
    const settings = {throttlingMethod: 'provided'};
    const context = {settings, computedCache: new Map()};
    const result = await MaxPotentialFID.request({trace, devtoolsLog, gatherContext, settings, URL},
      context);

    await expect(result).toMatchInlineSnapshot(`
Object {
  "timing": 16,
}
`);
  });
});
