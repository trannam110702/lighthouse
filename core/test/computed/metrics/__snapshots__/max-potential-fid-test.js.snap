// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics: Max Potential FID should compute a simulated value 1`] = `
Object {
  "optimistic": 50,
  "pessimistic": 50,
  "timing": 50,
}
`;
